File: android-app/core/model/src/main/kotlin/com/prayer/pointfinder/core/model/TileSources.kt
package com.prayer.pointfinder.core.model

object TileSources {
    private val STYLES = mapOf(
        "osm" to "https://tiles.openfreemap.org/styles/liberty",
        "osm-classic" to "https://pointfinder.pt/styles/osm-classic.json",
        "swisstopo" to "https://vectortiles.geo.admin.ch/styles/ch.swisstopo.basemap.vt/style.json",
        "swisstopo-sat" to "https://vectortiles.geo.admin.ch/styles/ch.swisstopo.imagerybasemap.vt/style.json",
        "voyager" to "https://basemaps.cartocdn.com/gl/voyager-gl-style/style.json",
        "positron" to "https://basemaps.cartocdn.com/gl/positron-gl-style/style.json",
    )

    const val DARK_STYLE_URL = "https://basemaps.cartocdn.com/gl/dark-matter-gl-style/style.json"

    private val DARK_STYLES = mapOf(
        "osm" to DARK_STYLE_URL,
        "voyager" to DARK_STYLE_URL,
        "positron" to DARK_STYLE_URL,
    )

    private val DEFAULT_CENTERS = mapOf(
        "osm" to Pair(40.08789650218038, -8.869461715221407),
        "osm-classic" to Pair(40.08789650218038, -8.869461715221407),
        "voyager" to Pair(40.08789650218038, -8.869461715221407),
        "positron" to Pair(40.08789650218038, -8.869461715221407),
        "swisstopo" to Pair(46.8182, 8.2275),
        "swisstopo-sat" to Pair(46.8182, 8.2275),
    )

    fun getDefaultCenter(key: String?): Pair<Double, Double> =
        DEFAULT_CENTERS[key] ?: DEFAULT_CENTERS["osm-classic"]!!

    fun getStyleUrl(key: String?): String =
        STYLES[key] ?: STYLES["osm-classic"]!!

    fun getResolvedStyleUrl(key: String?, isDark: Boolean): String {
        if (isDark) {
            val darkUrl = DARK_STYLES[key ?: "osm-classic"]
            if (darkUrl != null) return darkUrl
        }
        return getStyleUrl(key)
    }
}

File: ios-app/dbv-nfc-games/Utils/TileSources.swift
import Foundation
import CoreLocation

enum TileSources {
    private static let styles: [String: String] = [
        "osm": "https://tiles.openfreemap.org/styles/liberty",
        "osm-classic": "https://pointfinder.pt/styles/osm-classic.json",
        "swisstopo": "https://vectortiles.geo.admin.ch/styles/ch.swisstopo.basemap.vt/style.json",
        "swisstopo-sat": "https://vectortiles.geo.admin.ch/styles/ch.swisstopo.imagerybasemap.vt/style.json",
        "voyager": "https://basemaps.cartocdn.com/gl/voyager-gl-style/style.json",
        "positron": "https://basemaps.cartocdn.com/gl/positron-gl-style/style.json",
    ]

    private static let darkStyles: [String: String] = [
        "osm": "https://basemaps.cartocdn.com/gl/dark-matter-gl-style/style.json",
        "voyager": "https://basemaps.cartocdn.com/gl/dark-matter-gl-style/style.json",
        "positron": "https://basemaps.cartocdn.com/gl/dark-matter-gl-style/style.json",
    ]

    // Safe compile-time constant — guaranteed valid URL literal
    private static let defaultStyleURL = URL(string: "https://tile.openstreetmap.org/{z}/{x}/{y}.png")!

    static let darkStyleURL = URL(string: "https://basemaps.cartocdn.com/gl/dark-matter-gl-style/style.json")!

    static func styleURL(for key: String?) -> URL {
        let urlString = styles[key ?? "osm-classic"] ?? styles["osm-classic", default: ""]
        return URL(string: urlString) ?? Self.defaultStyleURL
    }

    static func resolvedStyleURL(for key: String?, isDark: Bool) -> URL {
        let resolvedKey = key ?? "osm-classic"
        if isDark, let darkURLString = darkStyles[resolvedKey] {
            return URL(string: darkURLString) ?? Self.defaultStyleURL
        }
        let urlString = styles[resolvedKey] ?? styles["osm-classic", default: ""]
        return URL(string: urlString) ?? Self.defaultStyleURL
    }

    private static let defaultCenters: [String: (lat: Double, lng: Double)] = [
        "osm": (40.08789650218038, -8.869461715221407),
        "osm-classic": (40.08789650218038, -8.869461715221407),
        "voyager": (40.08789650218038, -8.869461715221407),
        "positron": (40.08789650218038, -8.869461715221407),
        "swisstopo": (46.8182, 8.2275),
        "swisstopo-sat": (46.8182, 8.2275),
    ]

    static func defaultCenter(for key: String?) -> CLLocationCoordinate2D {
        let center = defaultCenters[key ?? "osm-classic"] ?? defaultCenters["osm-classic"]!
        return CLLocationCoordinate2D(latitude: center.lat, longitude: center.lng)
    }
}
